gdctools/lib/convert/tsv2magetab.py

```diff
@@ -3,33 +3,50 @@
 import csv
 from os.path import basename
 
-from ..common import safeMakeDirs, getTabFileHeader, map_blank_to_na, writeCsvFile
+from ..common import safeMakeDirs, getTabFileHeader, map_blank_to_na, writeCsvFile, rearrange_columns
 from ..meta import tcga_id, diced_file_paths
 
 
-def process(file_dict, infile, outdir, fpkm=False):
+def process(file_dict, infile, outdir, fpkm=False, col_scramble=None, data_cols=None):
+    '''
+    col_scramble is a list whose index is for the column in the output file and whose values 
+    represent column numbers of the input file.   Eg 0,2,3,4,1 will move column 1 of the input file to be after the others.  If col_scramble is None, column order is left unchanged.
+
+    Columns listed in data_cols get a sample name in the initial header row; 
+    header columns do not get labeled with a sample name.
+    If data_cols is None, assume all columns (after column 0) are data columns.
+ 
+    '''
+
+
     filepath = diced_file_paths(outdir, file_dict)[0]
     safeMakeDirs(outdir)
     _tcga_id = tcga_id(file_dict)
 
-    hdr1, hdr2 = generate_headers(infile, _tcga_id, fpkm)
+    hdr1, hdr2 = generate_headers(infile, _tcga_id, fpkm, data_cols)
+
 
     rawfile = open(infile, 'r')
     csvfile = csv.reader(fpkm_reader(rawfile) if fpkm else rawfile,
                          dialect='excel-tab')
 
     csvfile_with_hdr = change_header__generator(csvfile, hdr1, hdr2)
     csvfile_with_NAs = map_blank_to_na(csvfile_with_hdr)
+    csvfile_with_col_scramble = rearrange_columns(csvfile_with_NAs, col_scramble)
 
     safeMakeDirs(outdir)
-    writeCsvFile(filepath, csvfile_with_NAs)
+    writeCsvFile(filepath, csvfile_with_col_scramble)
 
     rawfile.close()
 
-def generate_headers(infile, tcga_id, fpkm):
+def generate_headers(infile, tcga_id, fpkm, data_cols):
     old_hdr = fpkm_header(infile).split() if fpkm else getTabFileHeader(infile)
-    num_data_cols = len(old_hdr) - 1
-    new_hdr = ['Hybridization REF'] + [tcga_id] * num_data_cols
+    new_hdr = ['Hybridization REF']
+    for i in range(1,len(old_hdr)):
+        if not data_cols or i in data_cols:
```

Comment:
```python
if data_cols is None or i in data_cols:
```
According to [PEP-8](https://www.python.org/dev/peps/pep-0008/#programming-recommendations), testing for `None` should be explicit.